client/src/components/Table/FilterComponent.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { Space, Checkbox } from 'antd';
+import { CheckboxChangeEvent } from 'antd/lib/checkbox';
+
+type Props = {
+  setHiddenColumnsRows: Function,
+  hiddenColumnsRows: Set<string>,
+  eventTypes: Array<string>,
+}
+
+const FilterComponent: React.FC<Props> = ({ setHiddenColumnsRows,  hiddenColumnsRows, eventTypes }) => {
+    const columnsName: Array<string> = ['Type', 'Special', 'Url', 'Organizer', 'Place'];
+    
+    const handledFilter = (event: CheckboxChangeEvent) => {
+        const {value} = event.target;
+        const {checked} = event.target;
+        if (checked && hiddenColumnsRows.has(value)) {
+          setHiddenColumnsRows((prevState: Set<string>) => {
```

Comment:
this looks a bit weird... This is a child component that implements a set flow for state from parent component. Let's simplify the flow and just call the parent function here and move all logic to parent component.